adapters/freewheelssp/freewheelssp.go

```diff
@@ -0,0 +1,119 @@
+package freewheelssp
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"golang.org/x/text/currency"
+	"net/http"
+)
+
+type RequestImpExt struct {
+	ZoneId int `json:"zoneId"`
+}
+
+type adapter struct {
+	endpoint string
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No impression in bid request.",
+		}}
+	}
```

Comment:
This check is not needed. The core logic of Prebid Server will never call `MakeRequests` without at least one impression.

Reply:
thanks, I removed this